### ISSUE TITLE
Deployments don't use customisable name

### ISSUE THREAD

**Issue by glasnt** [GoogleCloudPlatform/terraform-dynamic-python-webapp#63]
The variable deployment_name should be included in the deployment. 

`Name must start with a lowercase letter followed by up to 39 lowercase letters, numbers or hyphens, and cannot end with a hyphen`

**Comment by glasnt** [comment]
This should be done after #61 and #58 are merged, to minimize in flight changes. 

Initial proof of concept can be done now, there might just be some merge-magic required

**Comment by glasnt** [comment]
Borrow from terraform-**javascript**-webapp: https://github.com/GoogleCloudPlatform/terraform-dynamic-javascript-webapp/blob/v1.1.6/infra/variables.tf#L40

**Comment by glasnt** [comment]
Any future JSS should have this feature, but right now I think the complexity is out of scope. Closing.
